### ISSUE TITLE
OSX build looking for missing libtinfo

### ISSUE THREAD

**Issue by narc0tiq** [CleverRaven/Cataclysm-DDA#22435]
Though #22421 was fixed, we have a new problem as of [build 6932](http://gorgon.narc.ro:8080/job/Cataclysm-Matrix/6932/) ([OSX-Curses build log](http://gorgon.narc.ro:8080/job/Cataclysm-Matrix/6932/Graphics=Curses,Platform=OSX/console)).

My guess is that `pkg-config --libs ncurses` (probably) is adding `-ltinfo` to the `LDFLAGS`, which exists on the host system (obviously), but not in the OSX cross-compile environment.

I'm 99.9% sure this came about with #22333 changing out `HAVE_NCURSESW5CONFIG` for a `HAVE_PKGCONFIG`, but may also be related to making the change from `ncursesw` to `ncurses` in the same -- but fixing it (without just reverting that PR) is somewhat beyond me.

All I've been able to learn, after looking into this for a bloody *hour*, is that [the Makefile as of df38c299fa5849cf660d01bb20dc5efdce951741](https://github.com/CleverRaven/Cataclysm-DDA/blob/df38c299fa5849cf660d01bb20dc5efdce951741/Makefile) does correctly apply only `-lncurses`, as you can see in the [OSX-Curses log for build 6926](http://gorgon.narc.ro:8080/job/Cataclysm-Matrix/6926/Graphics=Curses,Platform=OSX/console), and I'm fairly certain that's coming from https://github.com/CleverRaven/Cataclysm-DDA/blob/df38c299fa5849cf660d01bb20dc5efdce951741/Makefile#L565-L567

In other words, the previous version would correctly ignore `ncursesw5-config` **even though it *is* available to the build machine**, meaning that somehow this block: https://github.com/CleverRaven/Cataclysm-DDA/blob/df38c299fa5849cf660d01bb20dc5efdce951741/Makefile#L554-L558 didn't set `HAVE_NCURSESW5CONFIG`. I haven't been able to follow the Makefile closely enough to figure out why that would be the case, and I'm out of time to dedicate to this.

**Comment by narc0tiq** [comment]
Let me explicitly point out that, as far as I can tell, this should've been broken already:
- `ncursesw5-config` is available on the build machine
- and returns the same results at `pkg-config ... ncursesw`
- which includes CFLAGS like `-I/usr/include/ncursesw` and LDFLAGS like `-ltinfo`

...and I just figured out it's because the check for `ncursesw5-config` is guarded by `ifeq ($(CROSS),)`, i.e., "if `$(CROSS)` is unset".

**Comment by Treah** [comment]
The problem is that ncurses5-config is being run instead of ncursesw5-config. This of course if the system even has this on it. Otherwise it will default to the -lncurses flag like it did before. It also could be that the pkg-config is using the wrong variable, instead of ncurses maybe it needs to be changed to ncursesw.

**Comment by narc0tiq** [comment]
Good thought, but actually the build machine has everything: `ncurses5-config`, `ncursesw5-config` and `pkg-config` . The problem is that none of them should be run when doing a **cross-compile**, because they all refer to the host, not the target.

It could be interesting to execute `$(CROSS)pkg-config`, but for osxcross doing so is a no-op -- that particular environment doesn't normally have ncurses at all. I believe @patternoia fed it in from an external source. And, of course, the Windows builds don't use ncurses, either.

**Comment by Treah** [comment]
I see the problem is when the function there falls though it will run the ncurses5-config when it should not and it should fall though to the -lncurses flag. hmmmm Yeah it should probably be put into that check to not do that if CROSS is set.

**Comment by narc0tiq** [comment]
Yup, it was very confusing to figure out in the first place, particularly since I so rarely have to read makefiles. Still, I think I got it with #22436.
